library/psa_crypto.c

```diff
@@ -1050,7 +1050,7 @@ psa_status_t psa_destroy_key( psa_key_handle_t handle )
 #endif /* MBEDTLS_PSA_CRYPTO_SE_C */
 
 #if defined(MBEDTLS_PSA_CRYPTO_STORAGE_C)
-    if( slot->attr.lifetime == PSA_KEY_LIFETIME_PERSISTENT )
+    if( slot->attr.lifetime != PSA_KEY_LIFETIME_VOLATILE )
```

Comment:
Could you please explain what difference this change makes?

Reply:
As the log message says:

> SE keys: make psa_destroy_key remove the key from storage

Before, for a key in a secure element (`lifetime` is neither `PERSISTENT` nor `VOLATILE`), the metadata wasn't removed, so the key object kept existing even though the key material was destroyed.